Frameworks/UIKit/UIView.mm

```diff
@@ -198,27 +199,68 @@ - (bool)_processGesturesForTouch:(UITouch*)touch event:(UIEvent*)event touchEven
         }
     }
 
-    // gesture priority list
-    const static id s_gesturesPriority[] = {[UIPinchGestureRecognizer class],
-                                            [UISwipeGestureRecognizer class],
-                                            [UIPanGestureRecognizer class],
-                                            [UILongPressGestureRecognizer class],
-                                            [UITapGestureRecognizer class] };
 
-    const static int s_numGestureTypes = sizeof(s_gesturesPriority) / sizeof(s_gesturesPriority[0]);
+    //
+    // Remove any gestures that should not simulatanously be recognized
+    //
 
-    //  Process all gestures
-    for (int i = 0; i < s_numGestureTypes; i++) {
-        id curgestureClass = s_gesturesPriority[i];
-        id gestures = [g_curGesturesDict objectForKey:curgestureClass];
-        if ([curgestureClass _fireGestures:gestures]) {
-            handled = true;
-            if (handled && DEBUG_GESTURES) {
-                TraceVerbose(TAG, L"Gesture (%hs) handled.", object_getClassName(curgestureClass));
+    // First find the active gesture
+    UIGestureRecognizer* activeGesture = nil;
+
+    int count = [g_currentlyTrackingGesturesList count];
+    for (int i = count - 1; i >= 0; i--) {
```

Comment:
@aballway I am guessing that you would prefer that we use a reverseObjectEnumerator to traverse in reverse order.  I have noticed that the code in this file uses a mixture of old for counting loops and for in loops.  Should all new code be for in when it can?  I initially used the style I found in the existing code.